src/ucs/stats/stats.c

```diff
@@ -460,6 +481,197 @@ void ucs_stats_node_free(ucs_stats_node_t *node)
+
+    pthread_mutex_lock(&ucs_stats_context.lock);
+
+    if (ucs_likely(node->cls->class_id != 0)) {
+        element = &ucs_array_elem(&ucs_stats_context.aggrgt_sum_db,
+                                  node->cls->class_id);
+        offset = element->aggregate_sum_class_offset;
+
+        goto offset_get_exit;
+    } else {
+        offset = ucs_stats_agrgt_sum_clsid_alloc(
+                         node, num_counters_in_class);
+    }
+
+offset_get_exit:
+    pthread_mutex_unlock(&ucs_stats_context.lock);
+
+    return offset;
+}
+
+
+/**
+ * Returns the aggregate_sum of all counters from all interfaces for a compact statistics trace.
+ * Used for Score-P UCX plugin to collect a compact trace of the counters.
+ *
+ * @param [in]  node                      Pointer to the statistics node.
+ * @param [in]  sel                       Whether or not to recursively iterate through the children.
+ * @param [in]  current_num_counters      Current number of counters iterated through (call with 0).
+ * @param [out] ucs_stats_aggregate_sum   The aggregate-sum of all counters of each class/type.
+ * @param [in]  num_stats_counters_max The number of elements in ucs_stats_aggregate_sum.
+ *
+ * Complexity: O(n)
+ *
+ * Returns: The total number of counters.
+ *
+ * Remarks: The function is recursive.
+ */
+static size_t ucs_stats_aggregate_sum_recurs(ucs_stats_node_t *node,
+                                             ucs_stats_children_sel_t sel,
+                                             size_t current_num_counters,
+                                             ucs_stats_counter_t *aggregate_sum,
+                                             size_t num_stats_counters_max)
+{
+    ucs_stats_filter_node_t *filter_node = node->filter_node;
+    size_t n_counters                    = current_num_counters;
+    ucs_stats_node_t *temp_node;
+    ucs_stats_node_t *child;
+    uint32_t counter_index;
+    uint32_t cnt_agrgt_sum_index;
+    uint32_t counter_output_index;
+    uint32_t num_counters_filtered;
+
+    /* Filter output */
+    counter_output_index  = 0;
+    num_counters_filtered = ucs_popcount(filter_node->counters_bitmask);
+    cnt_agrgt_sum_index   = 0;
+    ucs_for_each_bit(counter_index, filter_node->counters_bitmask) {
+        ucs_list_for_each(temp_node, &filter_node->type_list_head, type_list) {
+            cnt_agrgt_sum_index = counter_output_index +
+                                  ucs_stats_clsid_to_base_offset(
+                                          node, num_counters_filtered);
+
+            ucs_assert(cnt_agrgt_sum_index < num_stats_counters_max);
+            aggregate_sum[cnt_agrgt_sum_index] +=
+                    temp_node->counters[counter_index];
+        }
+        counter_output_index++;
+    }
+
+    n_counters += counter_output_index;
+
+    /* Children */
+    ucs_list_for_each(child, &node->children[sel], list) {
+        n_counters = ucs_stats_aggregate_sum_recurs(child, sel, n_counters,
```

Comment:
n_counters is overridden on every call, is this really then intent?

Reply:
Yes, the intent is to count recursively how many counters are in the list.

Reply:
1. you really have the last value, not total. it should have been +=
2. it's not even needed because there is global variable of number of counters in the list